contracts/apps/accounting/FinanceApp.sol

```diff
@@ -0,0 +1,432 @@
+            _approveTransaction(tid, "pre-approved payment");
+            p.nextTimestamp = next(p.ct_sec, p.ct_min, p.ct_hour, p.ct_day, p.ct_month, p.ct_weekday, p.ct_year, p.nextTimestamp);
+            p.timesCalled += 1;
+        }
+    }
+
+    /**
+    @return Returns the current accounting period id
+    */
+    function getCurrentAccountingPeriodId() public constant returns (uint){
+        // TODO: perhaps we should store the current accountingPeriod ID
+        // separately and allow accounting periods to be generated in advance.
+        // For now the current period is the most recent one
+        return accountingPeriods.length - 1;
+    }
+
+    /**
+    @dev This flattens the last TransactionUpdate with the base Transation to show the current state of the transaction.  This assumes that there is at least a single transaction update which is fine if newTransaction is used.
+    @param transactionId The id of the transaction 
+    @return The external addres, token address, amount, referncec string, transaction type, and current transaction state
+    */
+    function getTransactionInfo(uint transactionId) constant returns (address, address, uint, string, TransactionType, TransactionState) {
+        Transaction memory t = transactions[transactionId];
+        uint tuid = transactionUpdatesRelation[transactionId].length - 1;
+        uint lastTransactionUpdate = transactionUpdatesRelation[transactionId][tuid];
+        TransactionUpdate memory tu = transactionUpdates[lastTransactionUpdate];
+        return (t.externalAddress, t.token, t.amount, t.reference, t._type, tu.state);
+    }
+
+    /**
+    @dev Get the current state of a transaction
+    @param transactionId The id of the transaction 
+    @return Returns uint of the transactionstate and string of the reference
+    */
+    function getTransactionState(uint transactionId) constant returns (TransactionState, string) {
+        Transaction memory t = transactions[transactionId];
+        uint tuid = transactionUpdatesRelation[transactionId].length - 1;
+        uint lastTransactionUpdate = transactionUpdatesRelation[transactionId][tuid];
+        TransactionUpdate memory tu = transactionUpdates[lastTransactionUpdate];
+        return (tu.state, tu.reason);
+    }
+
+    /**
+    @dev Internal function to start the next accounting period
+    */
+    function _startNextAccountingPeriod() internal {
+        if(accountingPeriods.length == 0 || accountingPeriods[getCurrentAccountingPeriodId()].endTimestamp < now){
+            AccountingPeriod memory ap = defaultAccountingPeriodSettings;
+            ap.startTimestamp = now;
+            uint endTimestamp = next(ap.ct_sec, ap.ct_min, ap.ct_hour, ap.ct_day, ap.ct_month, ap.ct_weekday, ap.ct_year, now);
+            ap.endTimestamp = endTimestamp;
+            // TODO: store endBlock of last accountingPeriod?
+            ap.startBlock = block.number;
```

Comment:
Is startBlock used for anything? I would fire an event when the new Accounting Period starts and not store the block number